src/lib.rs

```diff
@@ -161,6 +182,9 @@ pub enum Error {
     TooManyHeaders,
     /// Invalid byte in HTTP version.
     Version,
+    #[cfg(feature = "utf8_in_path")]
+    /// Invalid UTF-8 in path.
+    Utf8Error,
```

Comment:
Unfortunately we cannot add a new error variant without a breaking change. But it's probably correct to just consider this a `Token` error.